### PR TITLE
[SB Redfish] Set Redfish system identifiers

### PR DESCRIPTION
**Pull request by jlongever**
These changes are to support active discovery of new Redfish System resources.

- Build MAC identifiers list using  System EthernetInterfaces resource members.
- Update unit-tests.

@RackHD/corecommitters @uppalk1 @tannoa2 @keedya @zyoung51

### CODE REVIEW

**Comment by jlongever** [comment]
test this please

**Comment by anhou** [comment]
:+1:

**Comment by anhou** [comment]
test this please

File: lib/jobs/redfish-discovery.js [hunk 1]
```diff
@@ -245,33 +246,67 @@ function RedfishDiscoveryJobFactory(
...
-            return self.upsertRelations(node, relations)
-            .then(function(nodeRecord) {
-                return Promise.all([
-                    waterline.obms.upsertByNode(nodeRecord.id, obm),
-                    nodeRecord
-                ]);
+            return self.redfish.clientRequest(config.root)
+            .then(function(res) {
+                var ethernet = _.get(res.body, 'EthernetInterfaces');
+                if(ethernet) {
```
Comment (yyscamper) [inline lib/jobs/redfish-discovery.js]: The `[]` is also a truth value, to enhance the judging criteria can prevent some extra and useless request from being sent if the `ethernet` is empty array.

File: lib/jobs/redfish-discovery.js [hunk 2]
```diff
@@ -245,33 +246,67 @@ function RedfishDiscoveryJobFactory(
...
+                    identifiers: identifiers
+                };
+                
+                var relations = [{
+                    relationType: 'enclosedBy',
+                    targets: targetList
+                }];
+
+                var obm = {
+                    config: config,
```
Comment (yyscamper) [inline lib/jobs/redfish-discovery.js]: A lot properties in the `config`, are they all necessary for redfish-obm-service? If not all, please only pick those necessary properties before upsert the obm.

File: lib/jobs/redfish-discovery.js (hunk 1 above)
Comment (yyscamper) [inline lib/jobs/redfish-discovery.js]: From the unit-testing code, the `EthernetInterfaces` is object rather than array, if so, ignore my above comment.

**Comment by yyscamper** [comment]
:+1: after addressing the comment of picking only necessary properties of redfish-obm-service.

**Comment by yyscamper** [comment]
Meanwhile, @jlongever , do you have ever consider to refactor the code to use RxJS? With so many powerful operators in RxJS, I think the code logic can be more simple. The long & nesting promise chains is a little hard to understand.

This is not a requirement for this PR, just raise some discussion.

**Comment by anhou** [comment]
test this please

File: lib/jobs/redfish-discovery.js (hunk 2 above)
Comment (jlongever) [inline lib/jobs/redfish-discovery.js]: Yes they are necessary.

File: lib/jobs/redfish-discovery.js (hunk 1 above)
Comment (jlongever) [inline lib/jobs/redfish-discovery.js]: Yes, Redfish spec defines EthernetInterfaces as an object type,   it should never be an array.

**Pull request merged**